src/queries/rum-pageviews.sql

```diff
@@ -1,137 +1,250 @@
---- description: Get daily page views for a site according to Helix RUM data
+--- description: Get page views for a given URL for a specified time period. You can control the reporting granularity and filter pageviews that have a specific checkpoint, source, and target.
 --- Authorization: none
 --- Access-Control-Allow-Origin: *
 --- interval: 30
 --- offset: 0
+--- startdate: 2020-01-01
+--- enddate: 2020-12-31
+--- checkpoint: -
+--- sources: -
+--- targets: -
 --- url: 
 --- granularity: 1
 --- timezone: UTC
 --- domainkey: secret
-DECLARE results NUMERIC;
-CREATE OR REPLACE PROCEDURE helix_rum.UPDATE_PAGEVIEWS (
-  ingranularity INT64,
-  inlimit INT64,
-  inoffset INT64,
-  inurl STRING,
-  intimezone STRING,
-  indomainkey STRING,
-  OUT results NUMERIC
-)
-BEGIN
-  CREATE TEMP TABLE temp_pageviews (
-    year INT64,
-    month INT64,
-    day INT64,
-    time STRING, --noqa: RF04
-    url INT64,
-    pageviews NUMERIC
+WITH dailydata AS (
+  SELECT
+    id,
+    checkpoint,
+    source,
+    target,
+    pageviews,
+    TIMESTAMP_TRUNC(time, DAY) AS trunc_date
+  FROM helix_rum.EVENTS_V3(
+    @url, # url
+    @offset, # offset
+    @interval, # days to fetch
+    @startdate, # start date
+    @enddate, # end date
+    @timezone, # timezone
+    'all', # deviceclass
+    @domainkey # domain key to prevent data sharing
+  )
+),
+
+weeklydata AS (
+  SELECT
+    id,
+    checkpoint,
+    source,
+    target,
+    pageviews,
+    TIMESTAMP_TRUNC(time, ISOWEEK) AS trunc_date
+  FROM helix_rum.EVENTS_V3(
+    @url, # url
+    @offset * 7, # offset in weeks
```

Comment:
This needs to be cast to int64 as Events_V3  expects in64, int64 for those two in the signature.